File: backend/app/schemas/briefing.py
"""Pydantic schemas for Morning Briefing API."""

from datetime import date, datetime
from typing import Any

from pydantic import BaseModel, Field


class BriefingTaskItem(BaseModel):
    """A single task suggested in a briefing."""

    task_id: str
    title: str
    priority: str
    reason: str | None = None


class BriefingResponse(BaseModel):
    """Response for a single briefing."""

    id: str
    briefing_date: date
    content: str
    tasks_suggested: list[BriefingTaskItem | dict[str, Any]] = Field(default_factory=list)
    wellbeing_snapshot: dict[str, Any] = Field(default_factory=dict)
    status: str
    read_at: datetime | None = None
    created_at: datetime


class BriefingHistoryResponse(BaseModel):
    """Response for briefing history."""

    briefings: list[BriefingResponse] = Field(default_factory=list)
    days: int = Field(ge=1, le=90)


class BrainDumpRequest(BaseModel):
    """Request for brain dump text processing."""

    text: str = Field(min_length=1, max_length=5000, description="Raw brain dump text")


class BrainDumpResponse(BaseModel):
    """Response after processing a brain dump."""

    tasks_created: int
    tasks: list[dict[str, Any]] = Field(default_factory=list)
    message: str

File: mobile/services/briefing.ts
import api from "./api";

export interface BriefingTaskItem {
  task_id: string;
  title: string;
  priority: string;
  reason: string | null;
}

export interface BriefingData {
  id: string;
  briefing_date: string;
  content: string;
  tasks_suggested: BriefingTaskItem[];
  wellbeing_snapshot: Record<string, number | string>;
  status: string;
  read_at: string | null;
  created_at: string;
}

export interface BrainDumpResult {
  tasks_created: number;
  tasks: Array<{ title: string; priority: string }>;
  message: string;
}

export interface BriefingHistoryResponse {
  briefings: BriefingData[];
  days: number;
}

export const briefingApi = {
  getToday: async (): Promise<BriefingData | null> => {
    const response = await api.get<BriefingData | null>("/briefing/today");
    return response.data;
  },

  generate: async (): Promise<BriefingData> => {
    const response = await api.post<BriefingData>("/briefing/generate");
    return response.data;
  },

  getHistory: async (days = 7): Promise<BriefingHistoryResponse> => {
    const response = await api.get<BriefingHistoryResponse>(
      `/briefing/history?days=${days}`
    );
    return response.data;
  },

  markAsRead: async (id: string): Promise<void> => {
    await api.put(`/briefing/${id}/read`);
  },

  brainDump: async (text: string): Promise<BrainDumpResult> => {
    const response = await api.post<BrainDumpResult>("/briefing/brain-dump", {
      text,
    });
    return response.data;
  },
};
